analyses/molecular-subtyping-LGAT/03-subset-cnv-files-LGAT.Rmd

```diff
@@ -0,0 +1,329 @@
+# Code used to generate pfamDataBioMart.RDS in https://gist.github.com/kgaonkar6/02b3fbcfeeddfa282a1cdf4803704794
+
+bioMartDataPfamFGFR1 <-
+  readRDS(system.file("extdata", "pfamDataBioMart.RDS", package = "annoFuse")) %>%
+  # to check if biospecimen harbors FGFR1 TKD (tyrosine kinase domain tandem duplication)
+  dplyr::filter(
+    hgnc_symbol == "FGFR1",
+    DESC == "Protein tyrosine kinase"
+  )
+```
+
+### Gather lgat subset
+
+```{r}
+# File from 00-LGAT-select-pathology-dx that is used for the pathology diagnosis
+# was used to subset clinical file
+lgat_specimens_df <- read_tsv(file.path(subset_dir, "lgat_metadata.tsv"))
+
+# Filter to RNA-Seq samples
+lgat_wgs_biospecimen_ids <- lgat_specimens_df %>%
+  dplyr::filter(experimental_strategy == "WGS") %>%
+  dplyr::pull(Kids_First_Biospecimen_ID)
+
+# Filter CNV calls
+manta_lgat <- manta %>%
+  dplyr::filter(Kids.First.Biospecimen.ID.Tumor %in% lgat_wgs_biospecimen_ids)
+
+cnv_lgat <- cnv %>%
+  dplyr::filter(ID %in% lgat_wgs_biospecimen_ids)
+```
+
+### Overlap tandem duplication in FGFR1 with exons 10-18 which make up the kinase domain, per [comment](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/790#issuecomment-729229244).
+
+First, create genomic ranges objects for CNV seg file and gene.
+
+```{r}
+# Genomic range for cnv seg file
+cnv_gr <- cnv %>%
+  # format
+  dplyr::rename(
+    chr = chrom, start = loc.start, end = loc.end,
+    copy_number = copy.num
+  ) %>%
+  dplyr::select(-num.mark, -seg.mean) %>%
+  # make genomic ranges
+  GenomicRanges::makeGRangesFromDataFrame(
+    keep.extra.columns = TRUE,
+    starts.in.df.are.0based = FALSE
+  )
+
+# Genomic range for gene location and domain overlap file
+domain_gr <- bioMartDataPfamFGFR1 %>%
+  # keep only domain info
+  dplyr::filter(!is.na(domain_start), !is.na(domain_end)) %>%
+  # format
+  dplyr::mutate(
+    strand = if_else(strand == "-1", "-", "+"),
+    domain_chr = paste0("chr", domain_chr)
+  ) %>%
+  dplyr::rename(chr = domain_chr, start = domain_start, end = domain_end) %>%
+  # make genomic ranges
+  GenomicRanges::makeGRangesFromDataFrame(
+    keep.extra.columns = TRUE,
+    starts.in.df.are.0based = FALSE,
+    seqnames.field = "chr"
+  )
+```
+
+
+
+### Distribution of manta SV
+SV should be annotated as tandem duplicate and overlap 
+"Protein tyrosine kinase" domain of FGFR1 in biospecimens
+
+```{r}
+
+# Genomic range for manta file
+manta_gr <- manta_lgat %>%
+  # format
+  dplyr::rename(
+    chr = SV.chrom, start = SV.start, end = SV.end,
+    ALT = ALT
+  ) %>%
+  # format with chr to match to cnv
+  dplyr::mutate(chr = paste0("chr", chr)) %>%
+  # make genomic ranges
+  GenomicRanges::makeGRangesFromDataFrame(
+    keep.extra.columns = TRUE,
+    starts.in.df.are.0based = FALSE
+  )
+
+# overlap manta and domain
+overlaps_manta <- IRanges::subsetByOverlaps(manta_gr, domain_gr)
```

Comment:
So this one is by default set to "any" overlap, which I don't think is what we want. We probably at the least should set a "minOverlap" parameter. (And we can probably base this on how long `domain_gr` is).

Reply:
good idea, I can add that param